### PR TITLE
Add support for confidential compute

### PR DESCRIPTION
**Pull request by eranco74**
Adds support for enabling confidential computing service on GCP
Feature link: https://issues.redhat.com/browse/OCPBU-142

https://github.com/openshift/api/pull/1384 - MERGED

Note, This PR updates the openshift/api and go vendoring so it brings along a few API changes.
To accommodate these changes I made the following updates:
 1. update openshift/cluster-api-provider-libvirt to latest master due to cluster-api update.
 2. replace deprecated sets.String with sets.Set[string].
 3. update github.com/golang/mock to v1.7.0-rc.1 to support generating mock for interfaces with generics.

### CODE REVIEW

**Comment by patrickdillon** [comment]
/approve
This LGTM but waiting for API merge

File: data/data/gcp/cluster/main.tf [hunk 1]
```diff
@@ -33,12 +33,14 @@ module "master" {
...
   zones           = distinct(var.gcp_master_availability_zones)
   secure_boot     = var.gcp_master_secure_boot
-
   root_volume_size = var.gcp_master_root_volume_size
   root_volume_type = var.gcp_master_root_volume_type
 
   root_volume_kms_key_link = var.gcp_root_volume_kms_key_link
 
+  confidential_compute = var.gcp_master_confidential_compute
+  on_host_maintenance         = var.gcp_master_on_host_maintenance
```
Comment (JoelSpeed) [inline data/data/gcp/cluster/main.tf:42]: Indentation off here

File: data/data/gcp/cluster/master/variables.tf [hunk 2]
```diff
@@ -73,3 +73,15 @@ variable "secure_boot" {
...
+  type        = string
+  description = "Defines whether the instance should have confidential compute enabled."
+  default     = ""
+}
+
+variable "on_host_maintenance" {
+  type        = string
+  description = "The behavior when a maintenance event occurs."
+  default     = ""
+}
```
Comment (JoelSpeed) [inline data/data/gcp/cluster/master/variables.tf:87]: Nit, files should always have a new line at the end, can cause weird git diffs if you don't

File: data/data/gcp/variables-gcp.tf [hunk 3]
```diff
@@ -151,3 +151,15 @@ variable "gcp_master_secure_boot" {
...
+  type = bool
+  description = "Defines whether the instance should have confidential compute enabled."
+  default = ""
+}
+
+variable "gcp_master_on_host_maintenance" {
+  type = string
+  description = "The behavior when a maintenance event occurs."
+  default = ""
+}
```
Comment (JoelSpeed) [inline data/data/gcp/variables-gcp.tf:165]: Nit, new line at end of file?

**Comment by eranco74** [comment]
e2e-azure-ovn is failing with this error:
```
level=error msg=, controlPlane.platform.azure.type: Invalid value: "Standard_D8s_v3": not found in region centralus, compute[0].platform.azure.type: Invalid value: "Standard_D4s_v3": not found in region centralus]
```
Seems to happen a lot and not related to this change
https://search.ci.openshift.org/?search=Invalid+value%3A+%22Standard_D8s_v3%22&maxAge=168h&context=1&type=bug%2Bissue%2Bjunit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

**Comment by eranco74** [comment]
/test e2e-gcp-ovn



<br class="Apple-interchange-newline">

**Comment by patrickdillon** [comment]
/override ci/prow/e2e-gcp-ovn 

GCP CI is currently broken. This test passed earlier on the same commit and the secureboot test has passed, so I feel ok to override.

**Pull request merged**